packages/angular_devkit/schematics/src/tree/action.ts

```diff
@@ -157,13 +157,16 @@ export function isContentAction(action: Action): action is CreateFileAction | Ov
 }
 
 
-export function isAction(action: any): action is Action {  // tslint:disable-line:no-any
-  const kind = action && action.kind;
+export function isAction(action: unknown): action is Action {
+  const act = action as Action;
```

Comment:
In all fairness I am not a big fan of this casting here, as this is hiding potential errors.

Reply:
Something like seems to be more "correct", but wondering if we should just stop at `hasOwnProperty` since it's the object has those 3 props it's should be an action.

```ts
export function isAction(action: unknown): action is Action {
  if (
    action
    && typeof action === 'object'
    && action.hasOwnProperty('path')
    && action.hasOwnProperty('id')
    && action.hasOwnProperty('kind')
  ) {
    const { kind, id, path } = action as { id: unknown, path: unknown, kind: unknown };
    if (
      typeof kind === 'string'
      && ['c', 'o', 'r', 'd'].includes(kind)
      && typeof id === 'number'
      && typeof path === 'string'
    ) {
      return true;
    }
  }

  return false;
}
```

//cc @clydin what do you think?

Reply:
based on the usage of the function, I think we should consider deprecating it and leaving it as-is for now.  I put together a PR that removes the only use internally: https://github.com/angular/angular-cli/pull/18637

Reply:
SGTM

Reply:
@santoshyadavdev, mind opening a separate PR to deprecate the this method?

Reply:
Ahh cool @alan-agius4 will do that today.

Reply:
@alan-agius4 please see https://github.com/angular/angular-cli/pull/18727, should I close this one.

Reply:
Thanks, I’ll close it for you.